src/lxcfs.c

```diff
@@ -770,27 +786,36 @@ static int lxcfs_access(const char *path, int mode)
 static int lxcfs_releasedir(const char *path, struct fuse_file_info *fi)
 {
 	int ret;
+	enum lxcfs_virt_t type;
 
-	if (strcmp(path, "/") == 0)
-		return 0;
+	type = file_info_type(fi);
 
-	if (strncmp(path, "/cgroup", 7) == 0) {
+	if (LXCFS_TYPE_CGROUP(type)) {
 		up_users();
 		ret = do_cg_releasedir(path, fi);
 		down_users();
 		return ret;
 	}
 
-	if (strcmp(path, "/proc") == 0)
-		return 0;
-
-	if (strncmp(path, "/sys", 4) == 0) {
+	if (LXCFS_TYPE_SYS(type)) {
 		up_users();
 		ret = do_sys_releasedir(path, fi);
 		down_users();
 		return ret;
 	}
 
+	if (LXCFS_TYPE_PROC(type))
```

Comment:
you don't need this check at all then. Cause for `/proc` directory we don't fill `fi` with any info in `lxcfs_opendir`:

```
static int lxcfs_opendir(const char *path, struct fuse_file_info *fi)
{
	int ret;

	if (strcmp(path, "/") == 0)
		return 0; // don't touch "fi"

	if (strncmp(path, "/cgroup", 7) == 0) {
		up_users();
		ret = do_cg_opendir(path, fi); // << fills "fi" with something
		down_users();
		return ret;
	}

	if (strcmp(path, "/proc") == 0)
		return 0; // don't touch "fi"

	if (strncmp(path, "/sys", 4) == 0) {
		up_users();
		ret = do_sys_opendir(path, fi); // << fills "fi" with something
		down_users();
		return ret;
	}

	return -ENOENT;
}
```

Reply:
Excellent point. I should try to extend my head-logic a bit further ;). Updated.